network/p2p/p2pbuilder/utils.go

```diff
@@ -21,3 +24,69 @@ func notEjectedPeerFilter(idProvider module.IdentityProvider) p2p.PeerFilter {
 		return nil
 	}
 }
+
+type limitConfigLogger struct {
+	logger zerolog.Logger
+}
+
+// newLimitConfigLogger creates a new limitConfigLogger.
+func newLimitConfigLogger(logger zerolog.Logger) *limitConfigLogger {
+	return &limitConfigLogger{logger: logger}
+}
+
+// withBaseLimit appends the base limit to the logger with the given prefix.
+func (l *limitConfigLogger) withBaseLimit(prefix string, baseLimit rcmgr.BaseLimit) {
+	l.logger = l.logger.With().
+		Int(fmt.Sprintf("%s_streams", prefix), baseLimit.Streams).
+		Int(fmt.Sprintf("%s_streams_inbound", prefix), baseLimit.StreamsInbound).
+		Int(fmt.Sprintf("%s_streams_outbound", prefix), baseLimit.StreamsOutbound).
+		Int(fmt.Sprintf("%s_conns", prefix), baseLimit.Conns).
+		Int(fmt.Sprintf("%s_conns_inbound", prefix), baseLimit.ConnsInbound).
+		Int(fmt.Sprintf("%s_conns_outbound", prefix), baseLimit.ConnsOutbound).
+		Int(fmt.Sprintf("%s_file_descriptors", prefix), baseLimit.FD).
+		Int64(fmt.Sprintf("%s_memory", prefix), baseLimit.Memory).Logger()
+}
+
+func (l *limitConfigLogger) loggerForLimits(config rcmgr.LimitConfig) zerolog.Logger {
```

Comment:
this looks like it may generate a very long log line. Have you run this on localnet to determine how long it will be? I'm worried it will get truncated somewhere in processing

Reply:
I did manual testing and the log prints fine, however, it is excessively long. Hence, I split it into several smaller logs: https://github.com/onflow/flow-go/pull/3904/commits/73462b35996046552fe157af5a277ec0e2e939e3